### PR TITLE
mullvad-vpn: support `aarch64-linux`

### PR DESCRIPTION
**Pull request by Enzime**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/mullvad-vpn/default.nix [hunk 1]
```diff
@@ -64,16 +64,26 @@ let
     systemd
   ];
 
+  version = "2023.3";
+
+  srcs = {
+    "x86_64-linux" = fetchurl {
+      url = "https://github.com/mullvad/mullvadvpn-app/releases/download/${version}/MullvadVPN-${version}_amd64.deb";
+      sha256 = "sha256-+XK9xUeSs93egmtsQ7qATug/n9taeQkmc4ZgObPYvn4=";
```
Comment (wegank) [inline pkgs/applications/networking/mullvad-vpn/default.nix:72]:
```suggestion
      hash = "sha256-+XK9xUeSs93egmtsQ7qATug/n9taeQkmc4ZgObPYvn4=";
```

File: pkgs/applications/networking/mullvad-vpn/default.nix [hunk 2]
```diff
@@ -64,16 +64,26 @@ let
...
+    };
+
+    "aarch64-linux" = fetchurl {
+      url = "https://github.com/mullvad/mullvadvpn-app/releases/download/${version}/MullvadVPN-${version}_arm64.deb";
+      sha256 = "sha256-19pr53exKUO3ACVvDA6afTKAG86mhI84hs/N/X2L3gk=";
```
Comment (wegank) [inline pkgs/applications/networking/mullvad-vpn/default.nix:77]:
```suggestion
      hash = "sha256-19pr53exKUO3ACVvDA6afTKAG86mhI84hs/N/X2L3gk=";
```

File: pkgs/applications/networking/mullvad-vpn/default.nix (hunk 1 above)
Comment (Enzime) [reply pkgs/applications/networking/mullvad-vpn/default.nix]: Updated :+1:

**Comment by wegank** [comment]
Would you like to add an update script as in #211642?

**Comment by Enzime** [comment]
@wegank updated :+1:

**Comment by Enzime** [comment]
@wegank I've rebased my PR :)

**Review by ymarkus — CHANGES_REQUESTED** [review]
~~I'm not sure if this belongs here...~~ see below
Otherwise seems good to me, but I haven't had time to test.

File: pkgs/development/compilers/fasmg/default.nix [hunk 3]
```diff
@@ -32,7 +32,7 @@ stdenv.mkDerivation rec {
         bin = "source/macos/fasmg";
         asm = "source/macos/fasmg.asm";
       };
-    }.${system} or (throw "Unsopported system: ${system}");
+    }.${system} or (throw "Unsupported system: ${system}");
```
Comment (ymarkus) [inline pkgs/development/compilers/fasmg/default.nix:35]: Sorry, meant this! :see_no_evil:

Comment (Enzime) [reply pkgs/development/compilers/fasmg/default.nix:35]: Would be overkill in my opinion to pull it out into another PR, could move it to a separate commit but I personally think it's just a typo so it's fine as is

**Review by wegank — APPROVED** [review]

**Pull request merged**